### PR TITLE
Add periodicity

### LINKED ISSUE #14: Handle periodicity
Add network edges across periodic boundaries to correctly handle crystals

### PR DESCRIPTION
**Pull request by streeve**
~~Still a work in progress. Breaks during forward propagation, but should add periodic atoms correctly~~
Adds periodic edges, but not periodic edge length (as weights)

Closes #14

### CODE REVIEW

**Comment by streeve** [comment]
This breaks because I added periodic atoms directly to the system, but did not update the network to handle them. This is also not what we want: the network should only be predicting for the real atoms (not ghosts). 

As discussed with @allaffa, I need to change the periodicity routine to build the `edge_index` directly (probably bypassing the adjacency matrix), adding unique ghost neighbors _that index back to the real atom they are based off of_.

We will leave the addition of the `edge_weights` to a separate PR to take into account atom distance directly

**Comment by streeve** [comment]
Fails as expected until #33 is taken care of

**Comment by streeve** [comment]
Very simple check right now, but at least includes the basic periodic test setup

File: run_config_input.py [hunk 1]
```diff
@@ -258,8 +253,6 @@ def run_normal_config_file(config_file="./examples/configuration.json"):
         model.__str__()
         + "-r-"
         + str(config["radius"])
-        + "-mnnn-"
-        + str(config["max_num_node_neighbours"])
```
Comment (pzhanggit) [inline run_config_input.py:262]: This probably is what causes the failure in testing. The model name in test_trained_model.py also needs updated/consistent with this one.

Comment (streeve) [reply run_config_input.py:262]: Thanks! Fixed

**Comment by streeve** [comment]
I think the accuracy failure here is related to removing the option to set maximum number of neighbors. Maybe that should be separated into a different PR

**Review by allaffa — APPROVED** [review]

**Comment by streeve** [comment]
Closing to be reopened in HydraGNN

**Pull request closed without merging**